server/chat-commands/moderation.ts

```diff
@@ -278,20 +287,20 @@ exports.commands = {
 
 	'!autojoin': true,
 	autojoin(target, room, user, connection) {
-		let targets = target.split(',');
+		const targets = target.split(',');
 		if (targets.length > 11 || connection.inRooms.size > 1) return;
 		Rooms.global.autojoinRooms(user, connection);
-		let autojoins = [];
+		const autojoins: string[] = [];
 
 		const promises = targets.map(
-			target => user.tryJoinRoom(target, connection).then(ret => {
+			roomid => user.tryJoinRoom(roomid as RoomID, connection).then(ret => {
 				if (ret === Rooms.RETRY_AFTER_LOGIN) {
-					autojoins.push(target);
+					autojoins.push(roomid);
 				}
 			})
 		);
 
-		Promise.all(promises).then(() => {
+		void Promise.all(promises).then(() => {
```

Comment:
```suggestion
		return Promise.all(promises).then(() => {
```

Like I said in your last PR, you should return them so the command handler can show the detailed crash instead of being swallowed up by the `unhandledRejection` handler.